Automattic/woocommerce-payments
# Allow UPE methods to be saved during Blocks checkout

Fixes #5260

#### Changes proposed in this Pull Request

This PR ensures that both shortcode and blocks checkouts are taken into account when determining whether the `save_payment_method` URL parameter is set. When saving payment methods from shortcode checkout, to check whether the save payment method checkbox is checked, we can use a variable name in a format similar to `wc-woocomerce_payments_sepa_debit-new-payment-method`, however in a blocks checkout, the gateway ID isn't present, leaving a variable in the format of `wc-sepa_debit-new-payment-method` in `$_POST` instead. 









#### Testing instructions

1. Remove all saved payment methods from your site.
2. From Blocks checkout, pay using SEPA and check the _Save payment information to my account for future purchases._ checkbox to save the payment method.
3. Visit **My Account > Payment Methods** and double check that the SEPA method that was just added appears on the page.
4. Repeat the above from shortcode checkout.
